src/wild_encounter.c

```diff
@@ -864,10 +878,84 @@ bool8 DoesCurrentMapHaveFishingMons(void)
         return FALSE;
 }
 
+static u32 GetLastFishingSpecies(void)
```

Comment:
Is there a reason there are so many functions that do only one thing?

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/4343#discussion_r1551680575

Reply:
1. If you end up using `GetLastFishingSpecies` in multiple places, but then the declaration /location / name of `sLastFishingSpecies` changes, you have to update it all over the place. With this layer of abstraction, you only need to change it in two places (Get and Set) and it will cascade out everywhere

2. Functions are generally supposed to do _one_ thing and do them well.

Reply:
> Functions are generally supposed to do one thing and do them well.

Not sure if I 100% agree. If we applied the rule to the whole expansion code base it would be hell to fix stuff because it would be hell to follow around. 

Obviously if there is a code block that is used multiply times (e.g. GetSideParty) it makes sense to make a function for it but if it's just 1 piece of code then I'm not sure.

Reply:
That's not a personal opinion I have, it's a very common coding practice and principle

> Functions should do one thing. They should do it well. They should do it only

Reply:
That principle assumes object-oriented programming and thereby doesn't necessarily apply here.